### PR TITLE
Upgrade rustfmt to 0.8.3

### PR DESCRIPTION
**Pull request by overvenus**
The currently used rustfmt is 0.6.0. It works great but it is outdated. It does not support new syntax `impl Trait`, which leads to CI failure if there are any `impl Trait` in the source code.

This PR upgrades rustfmt to 0.8.3. If you have any questions about the format style, please check out [rustfmt.toml](https://github.com/pingcap/tikv/pull/1749/commits/db28235137e0826c6601db8d2bff47997a150506#diff-1a80e1fce76d9ad7a4abc1720d360380), and leave comments about the options you want to change.

PTAL @siddontang @BusyJay @disksing @zhangjinpeng1987 @andelf @AndreMouche @nolouch @ngaut

### CODE REVIEW

File: benches/bin/mvcc.rs [hunk 1]
```diff
@@ -34,24 +34,28 @@ fn bench_tombstone_scan() -> BenchSamples {
 
     for (k, v) in kvs.take(100000) {
         let mut ts = ts_generator.next().unwrap();
-        store.prewrite(Context::new(),
+        store
+            .prewrite(Context::new(),
```
Comment (BusyJay) [inline benches/bin/mvcc.rs]: This is ugly, why start a new line?

File: src/bin/tikv-server.rs [hunk 2]
```diff
@@ -207,19 +210,28 @@ fn cfg_duration(target: &mut Duration, config: &toml::Value, name: &str) {
 }
 
 fn initial_log(matches: &Matches, config: &toml::Value) {
-    let level = get_flag_string(matches, "L")
-        .unwrap_or_else(|| get_toml_string(config, "server.log-level", Some("info".to_owned())));
-
-    let log_file_path = get_flag_string(matches, "f")
-        .unwrap_or_else(|| get_toml_string(config, "server.log-file", Some("".to_owned())));
+    let level = get_flag_string(matches, "L").unwrap_or_else(|| {
+                                                                 get_toml_string(config,
```
Comment (BusyJay) [inline src/bin/tikv-server.rs]: Too much indention.

File: src/raft/log_unstable.rs [hunk 3]
```diff
@@ -339,30 +327,46 @@ mod test {
...
-           (vec![new_entry(5, 1), new_entry(6, 1), new_entry(7, 1)], 5, None,
-               vec![new_entry(6, 2)],
-                   5, vec![new_entry(5, 1), new_entry(6, 2)]),
-
-            (vec![new_entry(5, 1), new_entry(6, 1), new_entry(7, 1)], 5, None,
-                vec![new_entry(7, 2), new_entry(8, 2)],
-                    5, vec![new_entry(5, 1), new_entry(6, 1), new_entry(7, 2), new_entry(8, 2)]),
-      ];
+        let tests = vec![// replace to the end
+                         (vec![new_entry(5, 1)],
```
Comment (BusyJay) [inline src/raft/log_unstable.rs]: This is sincerely unreadable.

**Comment by overvenus** [comment]
@BusyJay Thank you for you comments. For your suggestions, I can't agree with you more, and I have tried to keep the old style. As for your comments, could you help me find the options which are responsible for?

File: rustfmt.toml [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+# "Put spaces within non-empty parentheses"
+spaces_within_parens = false
+
+# "Replace uses of the try! macro by the ? shorthand"
+use_try_shorthand = false
+
+# "What Write Mode to use when none is supplied:
+# Replace, Overwrite, Display, Diff, Coverage"
+write_mode = "Replace"
```
Comment (siddontang) [inline rustfmt.toml]: I prefer overwrite here directly

File: rustfmt.toml [hunk 5]
```diff
@@ -0,0 +1,201 @@
+# "Use verbose output"
+verbose = false
```
Comment (siddontang) [inline rustfmt.toml]: do we need too many configurations?

File: rustfmt.toml (hunk 4 above)
Comment (overvenus) [inline rustfmt.toml]: Sure!

File: rustfmt.toml (hunk 5 above)
Comment (overvenus) [inline rustfmt.toml]: Actually, most of them are default values, they are here just for your information.

Comment (BusyJay) [inline rustfmt.toml]: Only list the configuration that are changed instead.

**Comment by overvenus** [comment]
Closed via #1750

**Pull request closed without merging**